src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build linux
+// +build linux
+
+package debug
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beevik/ntp"
+	"github.com/docker/go-units"
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system/syslog"
+	"gopkg.in/yaml.v2"
+)
+
+// Use the same date specs as journalctl (see `man journalctl`).
+const timeHelpText = `(journalctl date format, e.g. YYYY-MM-DD)`
+
+type step func() error
+
+type stepParams struct {
+	fs      afero.Fs
+	m       *sync.Mutex
```

Comment:
This can be `m sync.Mutex`.

Reply:
Since this gets copied over to each "`step`", wouldn't that cause the mutex instance to be copied too, resetting it for each?

Reply:
`stepParams` itself is initialized as a pointer, so the mutex itself is never copied. `go vet` would warn against mutex copies, which can help ensure this works (https://play.golang.org/).

Reply:
Done